protocol/client.go

```diff
@@ -0,0 +1,193 @@
+package protocol
+
+import (
+	"bytes"
+
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	m "github.com/coniks-sys/coniks-go/merkletree"
+)
+
+type ProofType int
+
+const (
+	InvalidProof ProofType = iota
+	ProofOfAbsence
+	ProofOfInclusion
+)
+
+// ConiksClient stores the current state of a CONIKS client.
+// This includes the current epoch and SignedTreeRoot value,
+// as well as directory's configurations
+// (e.g., wherether the TemporaryBinding extension is being used or not).
+// This struct also stores the verification results including
+// the type of returned proof.
+// The client should create a new ConiksClient instance only once,
+// when she registers her binding with a ConiksDirectory.
+// This ConiksClient instance then will be used to verify
+// the returned response from a ConiksDirectory.
+type ConiksClient struct {
+	CurrentEpoch uint64
+	SavedSTR     []byte
+
+	// verification results
+	ProofType          ProofType
+	VerificationResult ErrorCode
+
+	// extensions settings
+	isUseTBs bool
+	TBs      []*m.TemporaryBinding
+
+	// signing key
+	// TODO: should we pin the signing key and vrf key in the client? see #47
+	signKey sign.PublicKey
+}
+
+func NewConiksClient(savedSTR []byte, isUseTBs bool, signKey sign.PublicKey) *ConiksClient {
+	return &ConiksClient{
+		CurrentEpoch: 0,
+		SavedSTR:     savedSTR,
+		isUseTBs:     isUseTBs,
+		signKey:      signKey,
+	}
+}
+
+func (cs *ConiksClient) Verify(msg *Response, uname string, key []byte) {
```

Comment:
Could we add a brief comment the the `Verify` method, saying how the caller can identity if the verification has passed or not? (It looks like the caller would need to consult `VerificationResult`)